osu.Game/Utils/ModUtils.cs

```diff
@@ -333,5 +342,18 @@ public static bool IsValidFreeModForMatchType(Mod mod, MatchType type)
                     return mod.ValidForMultiplayerAsFreeMod;
             }
         }
+
+        /// <summary>
+        /// Determines whether a mod can be applied in the given freestyle mode.
+        /// </summary>
+        /// <param name="mod">The mod to test.</param>
+        /// <param name="freestyle">Whether freestyle is enabled.</param>
+        public static bool IsValidModForFreestyleMode(Mod mod, bool freestyle)
```

Comment:
I'd honestly much rather have this and `IsValidFreeModForMatchType()` become one method. I'm not sure it's ever valid to call one but not the other. Having them separate just feels like a footgun-in-waiting. And there's the added benefit that it will look closer to the web-side validation methods that way.

Reply:
Have merged all 3 methods with one that more closely mirrors the osu!web one.